src/main/scala/strawman/collection/mutable/Set.scala

```diff
@@ -48,6 +48,8 @@ trait SetOps[A, +CC[X], +C <: Set[A]]
     res
   }
 
+  def diff(that: collection.Set[A]): C = that.foldLeft(clone())(_ -= _)
```

Comment:
Maybe this would be better?

~~~ scala
coll.foldLeft(empty)((result, elem) => if (that.contains(elem)) result else result += elem)
~~~